include/fiction/utils/layout_utils.hpp

```diff
@@ -350,7 +351,56 @@ Lyt convert_to_fiction_coordinates(const sidb_cell_clk_lyt_siqad& lyt) noexcept
 
     return lyt_new;
 }
+/**
+ * Generates a random cell between two given cells.
+ *
+ * @tparam Lyt Cell-level layout type.
+ * @param cell1 First given cell.
+ * @return cell2 Second given cell.
+ */
+template <typename Lyt>
+typename Lyt::cell random_cell(const typename Lyt::cell& cell1, const typename Lyt::cell& cell2)
+{
+    typename Lyt::cell left_cell{};
+    typename Lyt::cell right_cell{};
+
+    if (cell1 <= cell2)
+    {
+        left_cell  = cell1;
+        right_cell = cell2;
+    }
+    else
+    {
+        left_cell  = cell2;
+        right_cell = cell1;
+    }
```

Comment:
Rather pass the variables by copy and use `std::swap` if the coordinates are in the "wrong" order

Reply:
Thank you! That's convenient.